breaking change: DockerCompose::start_service waits for service to come up

rukai opened the pull request:
Previously the `DockerCompose::new` constructor would wait properly but if we start a single service via `DockerCompose::start_service` no wait would occur.
This PR implements waiting in that scenario.

rukai committed:
DockerCompose::start_service waits for service to come up

cjrolo reviewed (approved):
Approved, but still working on understanding the long map/filter/etc with some chains inside which can make the code less readable for a future self.

rukai committed:
feedback from carlos

rukai committed:
Merge branch 'main' into start_service_wait

rukai commented:
Thanks for the feedback.
I moved the `Service` creation into a constructor which will hopefully reduce the noise and make the logic of `get_services` easier to follow.
I also added some comments to hopefully give a high level explanation of what the function does.
I'll go ahead and land this now.
